### ISSUE TITLE
Account details aren't emailed to the user when using the Stripe Checkout shortcode

### ISSUE THREAD

**Issue by mindctrl** [restrictcontentpro/restrict-content-pro#513]
When using the [register_form_stripe] shortcode, the account username and password aren't emailed to the user.

**Comment by mindctrl** [comment]
Instead of emailing the password, we talked about introducing a notice in the email, and possibly in the profile editor shortcode output, that shows a notice letting the user know they're using an auto-generated password. We can include a link in the email to the profile page so the user can change the password.

**Comment by pippinsplugins** [comment]
Moving this to 2.6.3.

**Comment by mindctrl** [comment]
Related to this, customer suggested a new email tag for sending the verification link in the active/free subscription email. Something like `%verification_link%`.

**Comment by mindctrl** [comment]
PR submitted. It attaches to the `rcp_stripe_signup` hook, checks that it's a Stripe Checkout shortcode signup, checks that it's a new user signup, hooks into the `retrieve_password_key` action hook that fires in WP core when a reset password key is generated, and rewrites the URLs in the email to use the RCP password reset form instead of wp-login.php.

I think in the future we can look at adding a setting to give admins more control over the text, once we have RCP_Emails in the wild for a release and when we start working on the settings overhaul.

I tested this with single site, the main site on multisite, and a sub site (subdirectory) on multisite.

Thoughts on this approach?

**Comment by pippinsplugins** [comment]
>>> Thoughts on this approach?

Thoughts? I love you. This is stellar!

@nosegraze could you test it please?

**Comment by nosegraze** [comment]
Works well in my test.

One thing we should make a note of in docs: It's using the `login_redirect` page but if someone doesn't have "Redirect Default Login URL" checked, then they might have the "Login Page" option to something that doesn't actually contain the shortcode, which means the reset password form won't load. We should just probably make a note of that in the Stripe Checkout docs.

Funnily enough that happened to me first try. XD

**Comment by mindctrl** [comment]
Ah yes, we can add a check for `hijack_login_url` which will lessen the chance of that happening, since if it's checked when settings are saved, it checks for the `[login_form]` (among a few others). I'll update the PR.

**Comment by mindctrl** [comment]
[Updated the PR](https://github.com/restrictcontentpro/restrict-content-pro/commit/f3d52e4f4b0ef070e70568a6f4c2fe9931da9ed9) to check for `hijack_login_url`. I also removed the search/replace of the wp-login.php appended to the bottom. The reason I did that is my regex-fu with preg_replace was not up to par and I couldn't get an exact match replace to work. Also, since we rewrite `login_url` with the hijack option, it's not as ugly.

If someone else wants to take a stab at that part, feel free.
